includes/utils.php

```diff
@@ -0,0 +1,307 @@
+        if ($block->attributes['queryId'] == $query_id) {
+          return $block;
+        }
+      }
+      if (count($block->innerBlocks ?? [])) {
+        $found = traverse($query_id, $block->innerBlocks);
+        if ($found) {
+          return $found;
+        }
+      }
+    }
+  }
+  return traverse($query_id, $mappedBlocks);
+}
+
+function get_mapped_blocks($post, $query_args)
+{
+  // get global styles from the theme.json file
+  $global_theme_styles = [];
+  $theme_directory = get_template_directory();
+  $theme_json_path = $theme_directory . '/theme.json';
+  if (file_exists($theme_json_path)) {
+    $theme_json_contents = file_get_contents($theme_json_path);
+    $theme_json_data = json_decode($theme_json_contents, true);
+    if($theme_json_data && $theme_json_data['styles'] && $theme_json_data['styles']['blocks']){
+      $global_theme_styles = $theme_json_data['styles']['blocks'];
+    }
+  }
+
+  // get global styles from db (these will be overrides from any default
+  // styles using the site editor, i.e. overriding any theme.json styles)
+  // `wp-global-styles-${themeName}`
+  $global_db_styles = [];
+  $theme_slug = get_stylesheet();
+  $global_styles_post = get_page_by_path("wp-global-styles-" . $theme_slug, OBJECT, 'wp_global_styles');
+  if ($global_styles_post && $global_styles_post->post_content) {
+    $decoded_global_styles = json_decode($global_styles_post->post_content, true);
+    if (isset($decoded_global_styles['styles']['blocks'])) {
+      $global_db_styles = $decoded_global_styles['styles']['blocks'];
+    }
+  }
+
+  $uri = $post->uri;
+  $main_blog_page_id = get_option('page_for_posts');
+  if ((!$post->ID && $uri === "/") || (!$post->ID && $main_blog_page_id !== "0") || ($post->ID == $main_blog_page_id)) {
+    // if no post id and the page is the index page,
+    // then we need to load the home.html template
+    // OR
+    // if there's no post id and there's a blog page set
+    // this is the main blog page so load the home.html template
+    // or if home not found, default to index
+    $page_template = get_block_template_by_slug("home");
+    if ($page_template) {
+      $templateBlocks = parse_blocks($page_template->content);
+    } else {
+      $page_template = get_block_template_by_slug("index");
+      if ($page_template) {
+        $templateBlocks = parse_blocks($page_template->content);
+      }
+    }
+  } else {
+    $the_post = get_post($post->ID);
+    $blocks = parse_blocks($the_post->post_content);
+
+    // Get the post meta values
+    $the_post_id = $the_post->ID;
+    $the_post_content = $the_post->post_content;
+    $the_post_type = $the_post->post_type;
+    $the_post_slug = $the_post->post_name;
+
+    $templateBlocks = get_template_blocks($the_post_id, $the_post_type, $the_post_slug);
+  }
+
+  // if there are no template blocks, i.e. no template was found, 
+  // then just set equal to the page blocks;
+  if (!$templateBlocks || !$query_args['postTemplate']) {
```

Comment:
All these kind of comments, if we can replace them with a semantic function, it would be great

Reply:
forget it, better be conservative in the context of this